flocker/node/agents/blockdevice.py

```diff
@@ -869,12 +878,12 @@ def run(self, deployer):
         )
         device = api.get_device_path(volume.blockdevice_id)
 
-        # This duplicates CreateFilesystem now.
-        check_output(["mkfs", "-t", "ext4", device.path])
+        create = CreateFilesystem(volume=volume, filesystem=u"ext4")
+        d = run_state_change(create, deployer)
 
         mount = MountBlockDevice(dataset_id=UUID(hex=self.dataset.dataset_id),
                                  mountpoint=self.mountpoint)
-        d = mount.run(deployer)
+        d.addCallback(lambda _: run_state_change(mount, deployer))
```

Comment:
Ah.  Because this mount operation wasn't being logged before but now it is.  And the logging assertion is getting confused and finding this unrelated mount event in the logs and treating it like it's the one it is looking for.

I guess that means `multistep_change_log` is broken.  It's finding children of actions other than the ones it is supposed to be finding.